nmmo/task/base_predicate.py

```diff
@@ -0,0 +1,430 @@
+# TODO: the below line will be gone after implementation
+# pylint: disable=unnecessary-pass
+from typing import List
+
+from nmmo.task.predicate import Predicate
+from nmmo.task.game_state import TeamGameState
+from nmmo.systems import skill as Skill
+from nmmo.lib.material import Material
+from nmmo.lib import utils
+
+from nmmo.entity.entity import EntityState
+
+
+class Timer(Predicate):
+  def __init__(self, num_tick: int):
+    super().__init__(num_tick)
+    self.num_tick = num_tick
+
+  def __call__(self, team_gs: TeamGameState, ent_id):
```

Comment:
Can this take GameState rather than TeamGameState?

Reply:
As we discussed, we will go with a global GameState instead of the current TeamGameState.

Reply:
Done!